libs/api/ebpf_api.cpp

```diff
@@ -1754,13 +1761,66 @@ ebpf_program_load_bytes(
     load_info.byte_code_size = byte_code_size;
     load_info.execution_context = execution_context_kernel_mode;
 
-    // TODO(issue #714): resolve maps in byte code.  Compare _resolve_maps_in_byte_code() in ebpfsvc.
-    load_info.map_count = 0;
+    // Resolve map handles in byte code.
+    std::vector<original_fd_handle_map_t> handle_map;
+    const ebpf_inst* instructions = reinterpret_cast<const ebpf_inst*>(byte_code);
+    const ebpf_inst* instruction_end = reinterpret_cast<const ebpf_inst*>(byte_code + byte_code_size);
+    for (size_t index = 0; index < byte_code_size / sizeof(ebpf_inst); index++) {
+        const ebpf_inst& first_instruction = instructions[index];
+        if (first_instruction.opcode != INST_OP_LDDW_IMM) {
+            continue;
+        }
+        if (&instructions[index + 1] >= instruction_end) {
+            return EBPF_INVALID_ARGUMENT;
+        }
+        index++;
+
+        // Check for LD_MAP flag.
+        if (first_instruction.src != 1) {
+            continue;
+        }
+
+        // Get the real map_fd value and handle.
+        int map_fd = static_cast<int>(first_instruction.imm);
+        ebpf_handle_t handle = Platform::_get_osfhandle(map_fd);
+
+        // Look up inner map id.
+        uint32_t type;
+        uint32_t key_size;
+        uint32_t value_size;
+        uint32_t max_entries;
+        ebpf_id_t inner_map_id;
+        result = query_map_definition(handle, &type, &key_size, &value_size, &max_entries, &inner_map_id);
+        if (result != EBPF_SUCCESS) {
+            continue;
```

Comment:
We could fail it, but it's not wrong behavior per se since the call to ebpf_rpc_load_program() at line 1815 will fail anyway.

Reply:
Since there is a continue stmt on failure, the outer map_fd will not get added to the handle_map. And the program load will certainly fail as the eBPF instruction is loading the fd. Since we know there is a 100% chance of failure, why not fail here without incurring the cost of RPC?

Reply:
ok done